File: cs430/SPLProjectB_PetroulesJ/Source/SPLProjectB_PetroulesJ_CS/SPLProjectB_PetroulesJ/InputHelper.cs
using System;

public class InputHelper
{
    /// <summary>
    /// Prompts the user to enter an integer greater than or equal to 0.
    /// </summary>
    /// <param name="promptMessage">The prompt message displayed to the user for input.</param>
    /// <param name="errorMessage">The error message displayed if the user does not enter an integer.</param>
    /// <returns>The integer that the user entered.</returns>
    public static int getInteger(string promptMessage, string errorMessage)
    {
        int id = -1;
        do
        {
            Console.Write(promptMessage);

            try
            {
                id = Convert.ToInt32(Console.ReadLine());
            }
            catch (FormatException)
            {
                Console.WriteLine(errorMessage);
            }
        }
        while (id == -1);

        return id;
    }

    public static int getIntegerInRange(string promptMessage, int min, int max)
    {
        bool valid = false;
        int id = 0;
        do
        {
            Console.Write(promptMessage);

            try
            {
                id = Convert.ToInt32(Console.ReadLine());
                valid = id >= min && id <= max;
            }
            catch (FormatException)
            {
                Console.WriteLine("Please enter a valid integer between " + min + " and " + max + ", inclusive.");
            }
        }
        while (!valid);

        return id;
    }

    public static String getString(string promptMessage, string errorMessage = null)
    {
        string name = string.Empty;
        do
        {
            Console.Write(promptMessage);
            name = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(name) && errorMessage != null)
            {
                Console.WriteLine(errorMessage);
            }
        }
        while (string.IsNullOrWhiteSpace(name));

        return name;
    }

    public static bool getBoolean(string promptMessage)
    {
        return InputHelper.getBoolean(promptMessage, "true", "false");
    }

    public static bool getBoolean(string promptMessage, string yes, string no)
    {
        string openClose = string.Empty;
        do
        {
            Console.Write(promptMessage);
            Console.Write("(enter " + yes + " or " + no + ") ");
            openClose = Console.ReadLine();
        }
        while (!string.Equals(openClose, yes, StringComparison.OrdinalIgnoreCase) && !string.Equals(openClose, no, StringComparison.OrdinalIgnoreCase));

        return string.Equals(openClose, yes, StringComparison.OrdinalIgnoreCase);
    }

    public static DateTime getDate(string promptMessage)
    {
        DateTime date = DateTime.MinValue;
        do
        {
            Console.Write(promptMessage);
            Console.Write("(enter yyyy-MM-dd) ");

            try
            {
                date = parseDate(Console.ReadLine());
            }
            catch (FormatException e)
            {
            }
        }
        while (date == DateTime.MinValue);

        return date;
    }

    public static DateTime parseDate(string date)
    {
        try
        {
            return DateTime.Parse(date);
        }
        catch (FormatException e)
        {
            throw e;
        }
    }
}

File: cs280/DSLib_PetroulesJ/src/dslib_petroulesj/InputHelper.java
package dslib_petroulesj;

import java.io.BufferedReader;
import java.io.IOException;
import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;

/**
 *
 * @author jakepetroules
 */
public class InputHelper
{
    /**
     * Prompts the user to enter an integer greater than or equal to 0.
     * @param reader The input stream to read from.
     * @param promptMessage The prompt message displayed to the user for input.
     * @param errorMessage The error message displayed if the user does not enter an integer.
     * @return The integer that the user entered.
     */
    public static int getInteger(BufferedReader reader, String promptMessage, String errorMessage) throws IOException
    {
        int id = -1;
        do
        {
            System.out.print(promptMessage);
            
            try
            {
                id = Integer.parseInt(reader.readLine());
            }
            catch (NumberFormatException ex)
            {
                System.out.println(errorMessage);
            }
        }
        while (id == -1);

        return id;
    }

    public static int getIntegerInRange(BufferedReader reader, String promptMessage, int min, int max) throws IOException
    {
        boolean valid = false;
        int id = 0;
        do
        {
            System.out.print(promptMessage);

            try
            {
                id = Integer.parseInt(reader.readLine());
                valid = id >= min && id <= max;
            }
            catch (NumberFormatException ex)
            {
                System.out.println("Please enter a valid integer between " + min + " and " + max + ", inclusive.");
            }
        }
        while (!valid);

        return id;
    }

    public static String getString(BufferedReader reader, String promptMessage) throws IOException
    {
        return getString(reader, promptMessage, null);
    }

    public static String getString(BufferedReader reader, String promptMessage, String errorMessage) throws IOException
    {
        String name = "";
        do
        {
            System.out.print(promptMessage);
            name = reader.readLine();
            if (name.equals("") && errorMessage != null)
            {
                System.out.println(errorMessage);
            }
        }
        while (name.equals(""));

        return name;
    }
    
    public static boolean getBoolean(BufferedReader reader, String promptMessage) throws IOException
    {
        return InputHelper.getBoolean(reader, promptMessage, "true", "false");
    }

    public static boolean getBoolean(BufferedReader reader, String promptMessage, String yes, String no) throws IOException
    {
        String openClose = "";
        do
        {
            System.out.print(promptMessage);
            System.out.print("(enter " + yes + " or " + no + ") ");
            openClose = reader.readLine();
        }
        while (!openClose.equalsIgnoreCase(yes) && !openClose.equalsIgnoreCase(no));

        return openClose.equalsIgnoreCase(yes);
    }

    public static Date getDate(BufferedReader reader, String promptMessage) throws IOException
    {
        Date date = null;
        do
        {
            System.out.print(promptMessage);
            System.out.print("(enter yyyy-MM-dd) ");

            try
            {
                date = parseDate(reader.readLine());
            }
            catch (ParseException e)
            {
            }
        }
        while (date == null);

        return date;
    }

    public static Date parseDate(String date) throws ParseException
    {
        DateFormat df = new SimpleDateFormat("yyyy-MM-dd");

        try
        {
            return df.parse(date);
        }
        catch (ParseException e)
        {
            throw e;
        }
    }
}
